### PR TITLE
feat: support adding labels and annotations to cluster secret (#7115)

### LINKED ISSUE #7115: "POST /cluster" API and Argo CD CLI should support adding labels and annotations
# Summary

Is should be possible to manage labels and annotations using "POST /cluster" API and Argo CD 

# Motivation

Cluster labels and annotations are useful to create applications using application set. Currently, labels and annotations have to be applied manually to cluster secret which is inconvenient and difficult to automate with k8s access.

# Proposal

Extend existing "POST /cluster" API and support manaing labels and annotations

### PR DESCRIPTION
**Pull request by chetan-rns**
Signed-off-by: Chetan Banavikalmutt <chetanrns1997@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

Introduces flags to add labels and annotations while creating cluster secrets. 

Example:
```shell
argocd cluster add api.server.example --label key=obj --annotation key2=obj2
```

Closes: #7115

### CODE REVIEW

**Comment by chetan-rns** [comment]
@alexmt I assumed we need this feature only while creating clusters. Should there be a provision to update them as well using `--upsert`?

File: pkg/apis/application/v1alpha1/types.go [hunk 1]
```diff
@@ -1275,6 +1275,10 @@ type Cluster struct {
 	ClusterResources bool `json:"clusterResources,omitempty" protobuf:"bytes,10,opt,name=clusterResources"`
 	// Reference between project and cluster that allow you automatically to be added as item inside Destinations project entity
 	Project string `json:"project,omitempty" protobuf:"bytes,11,opt,name=project"`
+	// Labels for cluster secret metadata
+	Labels map[string]string `json:"labels,omitempty" protobuf:"bytes,12,opt,name=labels"`
+	// Annotations for cluster secret metadata
+	Annotations map[string]string `json:"annotations,omitempty" protobuf:"bytes,13,opt,name=annotations"`
```
Comment (alexmt) [inline pkg/apis/application/v1alpha1/types.go:1281]: Please update Equals method in types.go as well:

https://github.com/argoproj/argo-cd/blob/bd7f8c8eda74c107b50df3908b3c6f8aff4be60d/pkg/apis/application/v1alpha1/types.go#L1281

It is used by Create API method to detect difference if cluster is already registered.

**Review by alexmt — CHANGES_REQUESTED** [review]
Thank you @chetan-rns ! Please address one comment about `Equals` method and it is good to go.

File: pkg/apis/application/v1alpha1/types.go (hunk 1 above)
Comment (chetan-rns) [reply pkg/apis/application/v1alpha1/types.go:1281]: Thanks! Updated Equals to return false on the mismatch of labels/annotations

**Review by alexmt — APPROVED** [review]
LGTM

**Pull request merged**